### PR TITLE
[Sketcher] Symmetry tool rework.

### PR DESCRIPTION
**Pull request by PaddleStroke**
Rework the symmetry tool, instead of using last selected geo as reference which is kind of clunky, you just select the geos you want and trigger the tool. Then you can hover on any line or point and see the resulting geos. Clicking on a line or point validate the tool.
You also now have the option to create symmetric constraints, so that the created geos are linked to the original.

image

Testing required. @FEA-eng

### CODE REVIEW

**Comment by ab0v3g4me** [comment]
I was just pondering with this today and it was weird that it doesn't create symmetric constraints by default :+1: for that.

What I don't get is how can you make symmetry relative to a point?

**Comment by FEA-eng** [comment]
@PaddleStroke I tested it, works very well. I just wonder - perhaps it shouldn't be possible to select both "Delete original geometries" and "Create symmetry constraints" at the same time since these options are mutually exclusive.

Also, it might be a good occasion to change the name of this feature. IMO it can be confused with symmetry constraints and should be named "Mirror" since that's also more intuitive.

Btw. do you plan to implement your chamfer tool too ? It's one of the last big missing features in the Sketcher now.

File: src/Mod/Sketcher/Gui/Resources/icons/pointers/Sketcher_Pointer_Create_Symmetry.svg
Comment (maxwxyz) [inline src/Mod/Sketcher/Gui/Resources/icons/pointers/Sketcher_Pointer_Create_Symmetry.svg:1]: Sketcher_Pointer_Create_Symmetry
@obelisk79 to be in line with the other pointer icons.

**Comment by PaddleStroke** [comment]
Small bug of exclusive checkboxes fixed (checking one unchecks the other).

Icon replaced.

**Comment by FEA-eng** [comment]
I tested it again, it's all good. Can you undraft it?

**Comment by yorikvanhavre** [comment]
Looks good! Merging

**Comment by 0penBrain** [comment]
Could one please point me to the GH issue where the problem solved by this PR was described, and the solution discussed ?

Also what is the current process applied by Ondsel to keep the documentation of FC aligned with its merged PRs ?

**Pull request merged**